src/osyris/spatial/spatial.py

```diff
@@ -0,0 +1,43 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Osyris contributors (https://github.com/nvaytet/osyris)
+
+import numpy as np
+
+
+def extract_sphere(dataset, radius, origin):
+    """
+    Extract a spherical subdomain around an origin point.
+    """
+    subdomain = dataset.__class__(nout=dataset.meta["nout"], path=dataset.meta["path"])
+    subdomain.meta = dataset.meta
+    subdomain._parent = dataset
```

Comment:
Why do you need to set this? I think Datasets have no `._parent` member, so you are effectively making the object different from a normal Dataset. Do you need this for something specific down the line?

Reply:
I added it to simply keep track of each subdomain's parent. This isn't going to be used in the module for now, but I have scripts where multiple datasets get loaded and it would be helpful to check which subdomain belongs to who.

Reply:
Hmmm, I still think it's weird to make the class different. Can't you work out the parent by the `nout` (output number) or something like that?